api/saml.go

```diff
@@ -0,0 +1,99 @@
+package api
+
+import (
+	"encoding/xml"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/crewjam/saml"
+	"github.com/crewjam/saml/samlsp"
+)
+
+// getSAMLServiceProvider generates a new service provider object with the
+// (optionally) provided descriptor (metadata) for the identity provider.
+func (a *API) getSAMLServiceProvider(identityProvider *saml.EntityDescriptor, idpInitiated bool) *saml.ServiceProvider {
+	externalURL, err := url.ParseRequestURI(a.config.API.ExternalURL)
+	if err != nil {
+		// this should not fail as a.config should have been validated using #Validate()
+		panic(err)
+	}
+
+	if !strings.HasSuffix(externalURL.Path, "/") {
+		externalURL.Path += "/"
+	}
+
+	externalURL.Path += "sso/"
```

Comment:
just wondering, why is this `sso/` and not `sso/saml/` ?

Reply:
The crewjam library adds `saml/` to the URLs. The tests cover for this if the library ever changes.